File: src/main/java/gr/apt/security/AuthenticationDto.java
package gr.apt.security;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import javax.validation.constraints.NotNull;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class AuthenticationDto {
    @NotNull
    private String username;
    @NotNull
    private String password;
}

File: src/main/java/gr/apt/lms/config/security/AuthenticationDto.kt
package gr.apt.lms.config.security

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import javax.validation.constraints.NotNull

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class AuthenticationDto(
    var username: @NotNull String = "",
    var password: @NotNull String = ""
)